src/Uns.ts

```diff
@@ -300,6 +296,22 @@ export default class Uns extends NamingService {
     return decoded[decoded.length - 1];
   }
 
+  async location(domain: string): Promise<DomainLocation> {
+    const [registry, resolver, owner] = await Promise.all([
+      this.registryAddress(domain),
```

Comment:
@DancingAxolotl You can use getData method and get `owner` and `resolver` in a single RPC call. https://github.com/unstoppabledomains/uns/blob/main/contracts/ProxyReader.sol#L151

Reply:
Thanks! I've updated the function